qubes/storage/reflink.py

```diff
@@ -227,14 +227,23 @@ def stop(self):
         else:
             if not self.snap_on_start:
                 self._size = self.size  # preserve manual resize of image
-            _remove_file(self._path_dirty)
-            _remove_file(self._path_clean)
+            try:
+                _remove_file(self._path_dirty)
+            except FileNotFoundError:
+                pass
+            try:
+                _remove_file(self._path_clean)
+            except FileNotFoundError:
+                pass
         return self
 
     def _commit(self, path_from):
         self._add_revision()
         self._prune_revisions()
-        qubes.utils.fsync_path(path_from)
+        try:
+            qubes.utils.fsync_path(path_from)
+        except FileNotFoundError:
+            return
```

Comment:
Better just prepend

```
if os.path.exists(self._path_dirty):
```

before the `_commit()` call in `stop()` instead of this. Otherwise the revision stuff at the beginning of `_commit()` would make it non-idempotent.

Reply:
Good point!  Fixed.